CMakeLists.txt

```
@@ -53,12 +53,15 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS Debug Release RelWithDebInfo MinSizeRel)
 endif()
 
+if (DEFINED BUILD_SHARED_LIBS AND NOT BUILD_SHARED_LIBS)
+  message(WARNING "You disabled BUILD_SHARED_LIBS, but preCICE unconditionally builds as shared library.")
```

```suggestion
  message(WARNING "You disabled BUILD_SHARED_LIBS, but preCICE can only being built as a shared library. This will still build preCICE as a shared library.")
```

---

> can only being built as a shared library.

This implies that the user is doing something wrong. In the case of a subproject, this is completely fine.

Slightly more verbose version:

```suggestion
  message(WARNING "You disabled BUILD_SHARED_LIBS. Note that 3.2.0 was the last release to honor this option. Newer preCICE releases always build as a shared library.")
```